src/App.js

```diff
@@ -2,22 +2,35 @@ import {
   BrowserRouter as  Router,
   Route,
   Switch,
-  Link,
+  Redirect,
 } from 'react-router-dom'
 import Welcome from './pages/Welcome/'
 import React from 'react'
 import Login from './pages/Login/'
 import Signup from './pages/Signup/'
-
 import './App.css';
 
+function PrivateRouteWelcome(props) {
+  const user = localStorage.getItem('user')
+
+  if(!user) return <Redirect to="/" />
+  return <Route {...props} />
+}
+
+function PrivateRouteLogin(props) {
+  const user = localStorage.getItem('user')
+
+  if(user) return <Redirect to="/welcome" />
+  return <Route {...props} />
+}
```

Comment:
Podrías usar un solo componente PrivateRoute que reciba un prop to y ahí decides a dónde lo redireccionas. No habría necesidad de tener 2 componentes iguales